workspaces/ui-v2/src/components/HttpBodySelector.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react';
+import {
+  FormControl,
+  FormControlLabel,
+  makeStyles,
+  Radio,
+  RadioGroup,
+} from '@material-ui/core';
+
+import { FontFamily } from '<src>/styles';
+
+type HttpBodySelectorProps<T> = {
+  items: T[];
+  children: (body: T) => React.ReactNode;
+  getDisplayName: (item: T) => string;
+};
+
+/**
+ * A component that is used to select / handle different items
+ */
+export const HttpBodySelector: <T extends {}>(
+  props: HttpBodySelectorProps<T>
+) => React.ReactElement = ({ items, children, getDisplayName }) => {
+  const [selectedItem, setSelectedItem] = useState(0);
+  const inBoundsSelected = Math.min(selectedItem, items.length - 1);
+  const classes = useStyles();
+
+  return (
+    <>
+      {items.length === 0 ? (
+        <>No bodies found</>
+      ) : items.length === 1 ? (
+        children(items[0])
+      ) : (
+        <>
+          <FormControl component="fieldset" className={classes.formGroup}>
+            <RadioGroup
+              value={selectedItem}
+              onChange={(e) => {
+                setSelectedItem(Number(e.target.value));
+              }}
+              className={classes.radioGroup}
+            >
+              {items.map((item, i) => (
+                <FormControlLabel
+                  classes={{
+                    label: classes.radio,
+                  }}
+                  key={getDisplayName(item)}
+                  value={i}
+                  control={<Radio />}
+                  label={getDisplayName(item)}
+                />
+              ))}
+            </RadioGroup>
+          </FormControl>
```

Comment:
I'm not sure radio buttons are what I would have gone with, but the reasons for that are probably too subtle to not just ship this for now (especially as our future bet is very likely to hit this UI again anyway). 

Just for completeness sake: in my mind, all use of form controls imply a user context where the user is making mutations to state. With these radio buttons you could interpret and read this as choosing which content type is currently active (not just for navigating, but in how it works). Once edit mode + other forms come into play, this issue becomes a bit bigger. With these buttons being used straight up for navigation purposes only, we could alleviate that source of confusion by rendering them with other types of elements.

Reply:
Hmmmmmm that's a really good point, and something I overlooked. Lets revisit this in our next bet - as this should be fairly easy to change since it's all in one space.

Reply:
+1 the @JaapRood feedback. Reviewing all PRs in tomorrow's release and this stood out to me. 

since we're touching this area of the UI next I say ship it.